packages/medusa/src/services/line-item.ts

```diff
@@ -378,6 +378,31 @@ class LineItemService extends BaseService {
     )
   }
 
+  /**
+   * Deletes a line item with the tax lines.
+   * @param {string} id - the id of the line item to delete
+   * @return {Promise<LineItem | undefined>} the result of the delete operation
+   */
+  async deleteWithTaxLines(id: string): Promise<LineItem | undefined> {
+    return await this.atomicPhase_(
+      async (transactionManager: EntityManager) => {
+        const lineItemRepository = transactionManager.getCustomRepository(
+          this.lineItemRepository_
+        )
+
+        const itemTaxLineRepo = transactionManager.getCustomRepository(
+          this.itemTaxLineRepo_
+        )
+
+        const lineItem = await this.retrieve(id)
+
+        await itemTaxLineRepo.delete({ item_id: id })
+
+        return await lineItemRepository.remove(lineItem)
```

Comment:
todo: can be replaced by `return await lineItemRepository.delete({ id: lineItem.id })`